app/layouts/InactivityDetect/InactivityDetect.tsx

```diff
@@ -42,6 +39,27 @@ const InactivityDetect: FC = () => {
     document.onkeypress = resetTimer;
   };
 
+  prepareForLogout = () => {
+    const networkBlockIndexBigInt = BigInt(selectedAccount.balanceStatus.networkBlockIndex);
+    const localBlockIndexBigInt = BigInt(selectedAccount.balanceStatus.localBlockIndex);
+    const accountBlockIndexBigInt = BigInt(selectedAccount.balanceStatus.accountBlockIndex);
+    const acceptableDiffBigInt = BigInt(2);
+    const isSynced =
+      networkBlockIndexBigInt >= accountBlockIndexBigInt &&
+      networkBlockIndexBigInt >= localBlockIndexBigInt &&
+      localBlockIndexBigInt >= accountBlockIndexBigInt &&
+      networkBlockIndexBigInt - accountBlockIndexBigInt < acceptableDiffBigInt;
+
+    if (isSynced) {
+      setInactiveTooLong(true);
+      document.onmousemove = null;
+      document.onkeypress = null;
+      reactionTimer = window.setTimeout(() => window.close(), TIME_FOR_REACTION);
+    } else {
+      reenableTimer();
+    }
+  };
+
```

Comment:
This logic should be extracted out; something similar is used in `SyncStatus`, and both could possibly use a same code.